### PR TITLE
feat: added predefined code to truncate prompt errors

### LINKED ISSUE #110: Support optional "parameters" field in function definitions
The following tool defintion must work.

```json
  {
    "tools": [
        {
            "type": "function",
            "function": {
                "name": "get_current_time",
                "description": "This function provides the current time"
            }
        }
    ]
  }

```

Related to https://github.com/epam/ai-dial-adapter-bedrock/issues/133

### PR DESCRIPTION
**Pull request by adubovik**
Blocked by https://github.com/epam/ai-dial-sdk/pull/151

Also accidentally resolves #110, because `sdk>=0.11.0` declares the tool parameters field optional, so we don't have to support the case when the field is missing to make the typechecker happy.

### CODE REVIEW

**Review by roman-romanov-o — DISMISSED** [review]

File: aidial_adapter_vertexai/chat/errors.py [hunk 1]
```diff
@@ -63,13 +60,7 @@ def __init__(self, message: str):
         super().__init__(self.message)
 
     def to_dial_exception(self) -> DialException:
-        return DialException(
-            status_code=422,
-            type="invalid_request_error",
-            message=self.message,
-            code="invalid_argument",
```
Comment (roman-romanov-o) [inline aidial_adapter_vertexai/chat/errors.py:70]: `code` parameter will change from "invalid_argument" to "400" after this change

File: aidial_adapter_vertexai/chat/errors.py [hunk 2]
```diff
@@ -34,13 +35,9 @@ async def report_usage(self, response: Response) -> None:
             await response.aflush()
 
     def to_dial_exception(self) -> DialException:
-        return DialException(
-            status_code=422,
-            type="invalid_request_error",
+        return RequestValidationError(
             message=self.error_message,
             display_message=self.error_message,
-            code="invalid_argument",
```
Comment (roman-romanov-o) [inline aidial_adapter_vertexai/chat/errors.py:42]: code parameter will change from "invalid_argument" to "400" after this change

Comment (adubovik) [reply aidial_adapter_vertexai/chat/errors.py:42]: good catch, fixed

File: aidial_adapter_vertexai/chat/errors.py (hunk 1 above)
Comment (adubovik) [reply aidial_adapter_vertexai/chat/errors.py:70]: good catch, fixed

**Review by roman-romanov-o — APPROVED** [review]

**Pull request merged**